ui/components/FormDialog.vue

```diff
@@ -0,0 +1,155 @@
+    @next="goToNextTab"
+  >
+    <template #sidebar>
+      <vertical-tabs
+        v-model:active-tab-key="activeTabKey"
+        :tabs="tabs"
+        :linear="mode === 'wizard'"
+      />
+    </template>
+    <template #default>
+      <slot :name="activeTabKey" />
+    </template>
+  </simple-dialog>
+</template>
+
+<script setup>
+import { ref, computed, watch } from 'vue';
+
+import SimpleDialog from '~/components/SimpleDialog.vue';
+import VerticalTabs from '~/components/VerticalTabs.vue';
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  tabs: {
+    type: Array,
+    required: true,
+  },
+  mode: {
+    type: String,
+    required: true,
+    validator(value) {
+      return ['wizard', 'edit'].includes(value);
+    },
+  },
+  form: {
+    type: Object,
+    required: true,
+  },
+  height: {
+    type: String,
+    default: '600px',
+  },
+  width: {
+    type: String,
+    default: '800px',
+  },
+  onSubmit: {
+    type: Function,
+    default: () => {},
+  },
+  detailsRoute: {
+    type: Object,
+    default: null,
+  },
+  rules: {
+    type: Object,
+    default: () => {},
+  },
+  submitLabel: {
+    type: String,
+    default: '',
+  },
+});
+
+const value = defineModel({
+  type: Boolean,
+  default: false,
+});
+
+const activeTabKey = ref('');
+const activeTabIdx = computed(() => props.tabs.findIndex((tab) => tab.key === activeTabKey.value));
+const activeTab = computed(() => props.tabs.find((tab) => tab.key === activeTabKey.value));
+const isFirstTab = computed(() => activeTabIdx.value === 0);
+const isLastTab = computed(() => activeTabIdx.value === props.tabs.length - 1);
+const isSummaryTab = computed(() => activeTabKey.value === 'summary');
+const isSubmittableTab = computed(() => activeTab.value.submittable);
+
+const computedTitle = computed(() => {
+  if (props.mode === 'edit') return props.title;
+  if (isSummaryTab.value) return `${props.title} – Summary`;
+  return `${props.title} – Step ${activeTabIdx.value + 1}`;
+});
+
+const isTabValid = computed(() => {
+  if (activeTab.value.includes) {
+    return activeTab.value.includes.every((field) => {
+      const fieldRules = props.rules[field];
+      if (fieldRules) {
+        return fieldRules.every((rule) => rule(props.form[field]));
+      }
+
+      return true;
+    });
+  }
+
+  return true;
+});
+
+const goToTab = (index) => {
+  activeTabKey.value = props.tabs[index].key;
+};
+const goToNextTab = () => {
+  const tabIndex = activeTabIdx.value + 1;
+  goToTab(tabIndex);
+};
+const goToPreviousTab = () => {
+  const tabIndex = activeTabIdx.value - 1;
+  goToTab(tabIndex);
+};
+
+const submit = async () => {
+  try {
+    await props.onSubmit();
+    if (props.mode === 'wizard' && !isLastTab.value) goToNextTab();
+  } catch (e) {
+    // empty
+    // TODO: Set text error in dialog?
+  }
+};
+
+const actions = computed(() => {
+  if (props.mode === 'edit') {
+    return ['spacer', 'close', 'save'];
```

Comment:
what is spacer?

Reply:
It's just an empty block to separate actions in dialogs, we also have it in the SPA.
See the space between "cancel" and "back", that's the spacer:
<kbd><img width="400" alt="image" src="https://github.com/cloudblue/connect-bi-reporter/assets/50662025/cbbb08b7-1ad6-405d-a4e5-c834bc9cfea5"></kbd>